### ISSUE TITLE
[array-type] array-simple does not allow T[] notation for generic types

### ISSUE THREAD

**Issue by sebthom** [typescript-eslint/typescript-eslint#11968]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.9.2&fileType=.tsx&code=C4TwDgpgBAsiAq5oF4oG8C%2BBuAsAKFElhAHEIA7CAJwEsBjRSAHgEEA%2BKVTXPfOge3IBnYFACGVKgEYAXMUYQA2gF1OUFVihQA9NqgB3flQDWQvoJHjJAJjlwylWgyRMRtcgHM2KtRq26oKjEaIQghKAAbGnJgaI8oaiojIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6AQ2mncNqNPRgA2uGw5EXAPbRIAGlFisY7JAAmiAGbsE%2BQZE7deySgFtiSSAuwBfKwF1Rt60A&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
type MyType1 = {};
const arr1: MyType1[] = [];  // works

type MyType2<A> = {};
const arr2: MyType2<string>[] = [];  // raises linting error
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/array-type": [ "error", {
        "default": "array-simple"
    }]
  }
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

Using [] array notation with predefined types and "array-simple" rule enabled should be allowed.

### Actual Result

A linting error is raised that this is not a simple type.

### Additional Info

**@typescript-eslint/array-type (array-simple)** treats generic type references as non-simple, contradicting docs (forces `ReadonlyArray<Foo<...>>` instead of `readonly Foo<...>[]`)

When configuring `@typescript-eslint/array-type` with `default: "array-simple"` and `readonly: "array-simple"`, the rule is documented to use `T[]` / `readonly T[]` for simple types, defined as *primitive names or type references*.

However, generic type references such as `MyType<A>` are treated as non-simple in practice, causing lint errors and (with autofix) rewriting to `Array<MyType<A>>`. This makes it impossible to enforce the desired style *readonly Foo[] for referenced types but ReadonlyArray<{...}> for inline object types* for generic references.

**Comment by bradzacher** [comment]
This is working as intended since the options introduction many years ago.

The intention is essentially that a simple type is just a single identifier, and everything is considered complex.
